### PR TITLE
requests: refactor initialization of a request app 

### PR DESCRIPTION
**Pull request by NRodriguezcuellar**
:heart: Thank you for your contribution!

partly closes https://github.com/inveniosoftware/invenio-communities/issues/491

Screenshot from 2022-04-05 10-25-03

### CODE REVIEW

File: invenio_requests/assets/semantic-ui/js/invenio_requests/request/actions/RequestActionModal.js [hunk 1]
```diff
@@ -37,18 +36,21 @@ export class RequestActionModal extends Component {
...
           open={modalOpen.modalId === modalId && modalOpen.isOpen}
           trigger={
-            <Button onClick={() => this.setOpen(true)} loading={loading}>
-              <Trans
-                defaults="{{action}}"
-                values={{ "action": action }}
-              />
-            </Button>
+            <Overridable
+              id="InvenioRequests.RequestActionModal.trigger"
```
Comment (kpsherva) [inline invenio_requests/assets/semantic-ui/js/invenio_requests/request/actions/RequestActionModal.js:40]: please revert this, it needs https://github.com/inveniosoftware/invenio-requests/issues/141

File: invenio_requests/assets/semantic-ui/js/invenio_requests/timeline/state/reducer.js [hunk 2]
```diff
@@ -12,15 +12,14 @@ import {
...
-const { size } = JSON.parse(requestDetailsDiv.dataset.defaultQueryConfig);
+
 
 export const initialState = {
   loading: false,
   refreshing: false,
   data: {},
   error: null,
-  size: size,
+  size: 15,
```
[the remark this answers is not in the document]
Comment (NRodriguezcuellar) [reply invenio_requests/assets/semantic-ui/js/invenio_requests/timeline/state/reducer.js:22]: note it being passed down via the redux store config now

File: invenio_requests/assets/semantic-ui/js/invenio_requests/request/actions/RequestActionModal.js (hunk 1 above)
Comment (NRodriguezcuellar) [reply invenio_requests/assets/semantic-ui/js/invenio_requests/request/actions/RequestActionModal.js]: done :+1:

File: invenio_requests/assets/semantic-ui/js/invenio_requests/InvenioRequestsApp.js [hunk 3]
```diff
@@ -56,9 +59,11 @@ InvenioRequestsApp.propTypes = {
   requestEventsApi: PropTypes.object,
   overriddenCmps: PropTypes.object,
   request: PropTypes.object.isRequired,
+  defaultQueryParams: PropTypes.object,
 };
 
 InvenioRequestsApp.defaultProps = {
   overriddenCmps: {},
   requestsApi: null,
+  defaultQueryParams: { size: 15 },
```
[the remark this answers is not in the document]
Comment (NRodriguezcuellar) [reply invenio_requests/assets/semantic-ui/js/invenio_requests/InvenioRequestsApp.js:68]: https://github.com/inveniosoftware/invenio-requests/pull/174/files#diff-f7a80612120744de5016c716c1a5651dc32df00c950ca5cfd9ef059758b1a3daR13

being passed down by the app configuration right here, these are just default props so nothing breaks in the case of it not being passed down

**Review by ntarocco — APPROVED** [review]

File: invenio_requests/assets/semantic-ui/js/invenio_requests/requestsAppInit.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import ReactDOM from "react-dom";
+import { InvenioRequestsApp } from "./InvenioRequestsApp";
+
+const requestDetailsDiv = document.getElementById("request-detail");
+const request = JSON.parse(requestDetailsDiv.dataset.record);
+const defaultQueryParams = JSON.parse(
+  requestDetailsDiv.dataset.defaultQueryConfig
+);
+
+const overriddenCmps = {
```
Comment (ntarocco) [inline invenio_requests/assets/semantic-ui/js/invenio_requests/requestsAppInit.js:17]: Minor: unused code?

**Review by kpsherva — APPROVED** [review]

**Pull request merged**